llvm/lib/IR/Metadata.cpp

```diff
@@ -1678,6 +1680,8 @@ void Instruction::dropUnknownNonDebugMetadata(ArrayRef<unsigned> KnownIDs) {
 
   // A DIAssignID attachment is debug metadata, don't drop it.
   KnownSet.insert(LLVMContext::MD_DIAssignID);
+  if (!ProfcheckDisableMetadataFixes)
+    KnownSet.insert(LLVMContext::MD_prof);
```

Comment:
I don't think that inserting it here is correct. This should be handled in the caller, like dropUBImplyingAttrsAndMetadata.